benchmark/util/priority-queue.js

```diff
@@ -0,0 +1,18 @@
+'use strict';
+
+const common = require('../common');
+
+const bench = common.createBenchmark(main, {
+  n: [1e6]
+}, { flags: ['--expose-internals'] });
+
+function main({ n, type }) {
+  const PriorityQueue = require('internal/priority_queue');
+  const queue = new PriorityQueue();
+  bench.start();
+  for (var i = 0; i < n; i++)
+    queue.insert(Math.random() * 1e7 | 0);
```

Comment:
@BridgeAR based on my testing, with 1e6 items, it ends up playing a very small role. I see extremely tiny variations run to run as a result but instead we'll be more likely to avoid someone over-optimizing this for the benchmark (which is very easy with a priority queue).